modules/services/muchsync.nix

```diff
@@ -0,0 +1,160 @@
+    enable = mkEnableOption "muchsync periodic sync";
+
+    frequency = mkOption {
+      type = types.str;
+      default = "*:0/5";
+      description = ''
+        How often to run muchsync. This
+        value is passed to the systemd timer configuration as the
+        <literal>OnCalendar</literal> option. See
+        <citerefentry>
+          <refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum>
+        </citerefentry>
+        for more information about the format.
+      '';
+    };
+
+    server = mkOption {
+      type = types.str;
+      description = ''
+        Remote SSH host to synchronize with.
+      '';
+    };
+
+    configFile = mkOption {
+      type = types.str;
+      default = "";
+      defaultText = "$NOTMUCH_CONFIG";
+      description = ''
+        Specify the path of the notmuch configuration file to use.
+        If none is specified, the default is to use the contents of the environment variable
+        $NOTMUCH_CONFIG, or if that variable is unset, the value $HOME/.notmuch-config.
+        (These are the same defaults as the notmuch command itself.)
+      '';
+    };
+
+    checkForModifiedFiles = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Check for modified files.
+        Without this option, muchsync assumes that files in a maildir are never edited.
+        -F disables certain optimizations so as to make muchsync at least check the
+        timestamp on every file, which will detect modified files at the cost of a longer startup time.
+        If muchsync dies with the error "message received does not match hash,"
+        you likely need to run it with the -F option.
+
+        Note that if your software regularly modifies the contents of mail files
+        (e.g., because you are running offlineimap with "synclabels = yes"),
+        then you will need to use -F each time you run muchsync.
+        Specify it as a server option (after the server name) if the editing happens server-side.
+      '';
+    };
+
+    remoteBinaryPath = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Specifies the path to muchsync on the server.
+        Ordinarily, muchsync should be in the default PATH on the server so this option is not required.
+        However, this option is useful if you have to install muchsync in a non-standard place or wish
+        to test development versions of the code.
+      '';
+    };
+
+    sshCommand = mkOption {
+      type = types.str;
+      default = "${pkgs.openssh}/bin/ssh -CTaxq";
+      defaultText = "ssh -CTaxq";
+      description = ''
+        Specifies a command line to pass to /bin/sh to execute a command on another machine.
+        The default value is "ssh -CTaxq".
+        Note that because this string is passed to the shell,
+        special characters including spaces may need to be escaped.
+      '';
+    };
+
+    noNew = mkOption {
```

Comment:
Please avoid inverted boolean options. I.e., if an option is set to `true` the feature should be enabled.

Reply:
Fixed.